reference/7.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Argument_Completion.md

```diff
@@ -0,0 +1,364 @@
+        $Type,
+        [Parameter(Mandatory=$true)]
+        [ArgumentCompleter( {
+            param ( $commandName,
+                    $parameterName,
+                    $wordToComplete,
+                    $commandAst,
+                    $fakeBoundParameters )
+
+            $possibleValues = @{
+                Fruits = @('Apple', 'Orange', 'Banana')
+                Vegetables = @('Tomato', 'Squash', 'Corn')
+            }
+            if ($fakeBoundParameters.ContainsKey('Type'))
+            {
+                $possibleValues[$fakeBoundParameters.Type] | Where-Object {
+                    $_ -like "$wordToComplete*"
+                }
+            }
+            else
+            {
+                $possibleValues.Values | ForEach-Object {$_}
+            }
+        } )]
+        $Value
+      )
+}
+```
+
+## ArgumentCompletions attribute
+
+The **ArgumentCompletions** attribute allows you to add tab completion values
+to a specific parameter. An **ArgumentCompletions** attribute must be defined
+for each parameter that needs tab completion. The **ArgumentCompletions**
+attribute is similar to **ValidateSet**. Both attributes take a list of values
+to be presented when the user presses <kbd>Tab</kbd> after the parameter name.
+However, unlike **ValidateSet**, the values are not validated. Therefore the
+user can supply any value, not just the values in the list.
+
+The **ArgumentCompletions** attribute should not be confused with the
+**ArgumentCompleter** attribute, which needs a scriptblock to define the
+options. the specified values are available
+
+The syntax is as follows:
+
+```powershell
+function Test-ArgumentCompletions {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$true)]
+        [ArgumentCompletions('Fruits', 'Vegetables')]
+        $Type,
+
+        [Parameter()]
+        [ArgumentCompletions('Apple', 'Banana', 'Orange')]
+        $Fruit,
+
+        [Parameter()]
+        [ArgumentCompletions('Tomato', 'Corn', 'Squash')]
+        $Vegetable
+    )
+}
+```
+
+Each of the parameters is provided a list of options to the
+**ArgumentCompletions** attribute to enable tab completion.
+
+This attribute was introduced in PowerShell Core 6.0
+
+## Class-based argument completers
+
+Beginning in PowerShell 7.2, a new feature was added that allows you to define
+more generic implementations of parameterized `ArgumentCompleters`.
+
+By deriving from `ArgumentCompleterAttribute`, it's possible to create generic
+completers like:
+
+```powershell
+[DirectoryCompleter(ContainingFile="pswh.exe", Depth=2)]
+
+[DateCompleter(WeekDay='Monday', From="LastYear")]
+
+[GitCommits(Branch='release')]
+```
+
+The derived attributes must implement the `IArgumentCompleterFactory` interface
+and use property values to create a specialized completer.
+
+```powershell
+using namespace System.Management.Automation
+using namespace System.Management.Automation.Language
+using namespace System.Collections
```

Comment:
Best practice is to have these in alphabetical order

Reply:
Fixed